### PR TITLE
Support export database with schema only

### LINKED ISSUE #5563: Feature: DDL-only export command
### API

Other

### Description

I've had a few instances while poking through some foreign databases, where I've wished for an equivalent command to SQLite's `.schema` or PostgreSQL's `pg_dump --schema-only` feature.   `show_tables()` and `show_connections()` are helpful, but getting the raw schema, with fields and types, in ascii form would certainly be nice for adding to a VCS/diffing, or just gaining understanding of layout quickly.

@prrao87 pointed me to `EXPORT DATABASE`, which generates a "schema.cypher" file, which is exactly what I'm after -- just without the overhead of the data export!

Perhaps another option to export that omits everything but the DDL?

```
EXPORT DATABASE '/path/to/export' (schema_only=true);
```

... or some such?

### PR DESCRIPTION
**Pull request by acquamarin**
Closes #5563 
This PR adds a new option: `schema_only` to the `export database` command. This option allows user to only export schemas.

### CODE REVIEW

File: src/include/planner/planner.h [hunk 1]
```diff
@@ -93,6 +93,8 @@ class KUZU_API Planner {
         binder::expression_vector outExprs);
 
     // Plan export/import database
+    void planExportTableData(const binder::BoundStatement& boundExportDatabase,
```
Comment (andyfengHKU) [inline src/include/planner/planner.h:96]: I would make this return a vector of operator rather than pass by reference.

File: test/test_files/copy/export_import_db.test [hunk 2]
```diff
@@ -288,3 +288,28 @@ Runtime exception: Import database failed: Binder exception: Table org does not
 -STATEMENT MATCH (a:`N Table`)-[e:`R Table`]->(b:`N Table`) RETURN a.*, e.*, b.*;
 ---- 1
 0|A|Edge|1|B
+
+-CASE ExportDBSchemaOnly
+-SKIP_WASM
+-SKIP_IN_MEM
+-STATEMENT EXPORT DATABASE "${KUZU_EXPORT_DB_DIRECTORY}_special/special_property_name" (SCHEMA_ONLY=true, format='csv');
```
Comment (andyfengHKU) [inline test/test_files/copy/export_import_db.test:295]: Add a test with empty database as well.

**Review by andyfengHKU — APPROVED** [review]
